src/zcl_abapgit_folder_logic.clas.abap

```diff
@@ -163,18 +164,31 @@ CLASS ZCL_ABAPGIT_FOLDER_LOGIC IMPLEMENTATION.
 
       CASE io_dot->get_folder_logic( ).
         WHEN zif_abapgit_dot_abapgit=>c_folder_logic-full.
-          rv_package = lv_new.
-          TRANSLATE rv_package USING '#/'.
+          lv_absolute_name = lv_new.
+          TRANSLATE lv_absolute_name USING '#/'.
           IF iv_top(1) = '$'.
-            CONCATENATE '$' rv_package INTO rv_package.
+            CONCATENATE '$' lv_absolute_name INTO lv_absolute_name.
           ENDIF.
         WHEN zif_abapgit_dot_abapgit=>c_folder_logic-prefix.
-          CONCATENATE rv_package '_' lv_new INTO rv_package.
+          CONCATENATE rv_package '_' lv_new INTO lv_absolute_name.
         WHEN OTHERS.
           ASSERT 0 = 1.
       ENDCASE.
 
-      TRANSLATE rv_package TO UPPER CASE.
+      TRANSLATE lv_absolute_name TO UPPER CASE.
+
+      IF strlen( lv_absolute_name ) > 30.
+        zcx_abapgit_exception=>raise( |Package { lv_absolute_name } exceeds ABAP 30-characters-name limit| ).
+      ENDIF.
+
+      IF lv_absolute_name = lv_top.
+        zcx_abapgit_exception=>raise( |Root package { lv_top } has a subpackage with the same name| ).
+      ENDIF.
+      IF lv_absolute_name = lv_parent.
```

Comment:
Guess this only checks the direct parent? And not grandparents? Anyhow its better than before

Reply:
Hm, true. 
What if I create a table with unique package names and then compare to it?

Reply:
yeah